RecoTracker/MkFitCore/src/KalmanUtilsMPlex.cc

```diff
@@ -0,0 +1,917 @@
+    T* c = C.fArray;
+    ASSUME_ALIGNED(c, 64);
+
+#pragma omp simd
+    for (int n = 0; n < N; ++n) {
+      c[0 * N + n] = a[0 * N + n] * b[0 * N + n] + a[1 * N + n] * b[3 * N + n] + a[3 * N + n] * b[6 * N + n];
+      c[1 * N + n] = a[0 * N + n] * b[1 * N + n] + a[1 * N + n] * b[4 * N + n] + a[3 * N + n] * b[7 * N + n];
+      c[2 * N + n] = 0;
+      c[3 * N + n] = a[1 * N + n] * b[0 * N + n] + a[2 * N + n] * b[3 * N + n] + a[4 * N + n] * b[6 * N + n];
+      c[4 * N + n] = a[1 * N + n] * b[1 * N + n] + a[2 * N + n] * b[4 * N + n] + a[4 * N + n] * b[7 * N + n];
+      c[5 * N + n] = 0;
+      c[6 * N + n] = a[3 * N + n] * b[0 * N + n] + a[4 * N + n] * b[3 * N + n] + a[5 * N + n] * b[6 * N + n];
+      c[7 * N + n] = a[3 * N + n] * b[1 * N + n] + a[4 * N + n] * b[4 * N + n] + a[5 * N + n] * b[7 * N + n];
+      c[8 * N + n] = 0;
+      c[9 * N + n] = a[6 * N + n] * b[0 * N + n] + a[7 * N + n] * b[3 * N + n] + a[8 * N + n] * b[6 * N + n];
+      c[10 * N + n] = a[6 * N + n] * b[1 * N + n] + a[7 * N + n] * b[4 * N + n] + a[8 * N + n] * b[7 * N + n];
+      c[11 * N + n] = 0;
+      c[12 * N + n] = a[10 * N + n] * b[0 * N + n] + a[11 * N + n] * b[3 * N + n] + a[12 * N + n] * b[6 * N + n];
+      c[13 * N + n] = a[10 * N + n] * b[1 * N + n] + a[11 * N + n] * b[4 * N + n] + a[12 * N + n] * b[7 * N + n];
+      c[14 * N + n] = 0;
+      c[15 * N + n] = a[15 * N + n] * b[0 * N + n] + a[16 * N + n] * b[3 * N + n] + a[17 * N + n] * b[6 * N + n];
+      c[16 * N + n] = a[15 * N + n] * b[1 * N + n] + a[16 * N + n] * b[4 * N + n] + a[17 * N + n] * b[7 * N + n];
+      c[17 * N + n] = 0;
+    }
+  }
+
+  void KalmanGain(const MPlexLS& A, const MPlex2S& B, MPlexL2& C) {
+    // C = A * B, C is 6x2, A is 6x6 sym , B is 2x2
+
+    typedef float T;
+    const idx_t N = NN;
+
+    const T* a = A.fArray;
+    ASSUME_ALIGNED(a, 64);
+    const T* b = B.fArray;
+    ASSUME_ALIGNED(b, 64);
+    T* c = C.fArray;
+    ASSUME_ALIGNED(c, 64);
+
+#include "KalmanGain62.ah"
+  }
+
+  inline void KHMult(const MPlexLH& A, const MPlexQF& B00, const MPlexQF& B01, MPlexLL& C) {
+    // C = A * B, C is 6x6, A is 6x3 , B is 3x6
+    KHMult_imp(A, B00, B01, C, 0, NN);
+  }
+
+  inline void KHC(const MPlexLL& A, const MPlexLS& B, MPlexLS& C) {
+    // C = A * B, C is 6x6, A is 6x6 , B is 6x6 sym
+
+    typedef float T;
+    const idx_t N = NN;
+
+    const T* a = A.fArray;
+    ASSUME_ALIGNED(a, 64);
+    const T* b = B.fArray;
+    ASSUME_ALIGNED(b, 64);
+    T* c = C.fArray;
+    ASSUME_ALIGNED(c, 64);
+
+#include "KHC.ah"
+  }
+
+  inline void KHC(const MPlexL2& A, const MPlexLS& B, MPlexLS& C) {
+    // C = A * B, C is 6x6 sym, A is 6x2 , B is 6x6 sym
+
+    typedef float T;
+    const idx_t N = NN;
+
+    const T* a = A.fArray;
+    ASSUME_ALIGNED(a, 64);
+    const T* b = B.fArray;
+    ASSUME_ALIGNED(b, 64);
+    T* c = C.fArray;
+    ASSUME_ALIGNED(c, 64);
+
+#include "K62HC.ah"
+  }
+
+  // //Warning: MultFull is not vectorized, use only for testing!
```

Comment:
Addressed in Round 3 follow-up PR trackreco/cmssw#78 commit https://github.com/trackreco/cmssw/pull/78/commits/ed82f6fae7816eeb9d7f941e1b1a864f55027acd
the code was uncommented, in case it is useful